lib/backend/k8s/resources/watcher.go

```diff
@@ -91,43 +102,45 @@ func (crw *k8sWatcherConverter) processK8sEvents() {
 	for {
 		select {
 		case event, ok := <-crw.k8sWatch.ResultChan():
-			var e *api.WatchEvent
+			var events []*api.WatchEvent
 			if !ok {
 				// The channel is closed so send a terminating watcher event indicating the watch was
 				// closed by the remote.
 				crw.logCxt.Debug("Watcher terminated by remote")
-				e = &api.WatchEvent{
+				events = []*api.WatchEvent{{
 					Type: api.WatchError,
 					Error: cerrors.ErrorWatchTerminated{
 						Err:            fmt.Errorf("terminating error event from Kubernetes watcher: closed by remote"),
 						ClosedByRemote: true,
 					},
-				}
+				}}
 			} else {
 				// We have a valid event, so convert it.
-				e = crw.convertEvent(event)
-				if e == nil {
+				events = crw.convertEvent(event)
+				if events == nil {
```

Comment:
Should we do `len(events) == 0` here? Might be safer.

Reply:
I believe the main thing this branch covered were cases like this: https://github.com/projectcalico/libcalico-go/blob/master/lib/backend/k8s/resources/workloadendpoint.go#L265

Does it still do that properly? I think probably not because it will be converted into a slice with a single entry here: https://github.com/projectcalico/libcalico-go/pull/1208/files#diff-a77ab7c88a05a1a2822365096c0ed0fdR51

Is that right?

Reply:
Think I resolved this bug now, added tests for it too